src/App/Fossa/Config/Report.hs

```diff
@@ -40,20 +43,45 @@ import Options.Applicative (
   option,
   optional,
   progDesc,
+  strOption,
   switch,
  )
 import Path.IO (getCurrentDir)
+import Text.Read (readMaybe)
 
 data ReportType = Attribution deriving (Eq, Ord, Enum, Bounded)
 
 instance Show ReportType where
   show Attribution = "attribution"
 
--- TODO: Add support for text-format reports
 data ReportOutputFormat
   = ReportJson
-  -- ReportPretty
-  deriving (Eq, Ord, Show)
+  | ReportMarkdown
+  | ReportSpdx
+  deriving (Eq, Ord, Enum, Bounded)
+
+instance Read ReportOutputFormat where
+  readsPrec _ s | s == show ReportJson = [(ReportJson, safeStripPrefix (show ReportJson) s)]
+  readsPrec _ s | s == show ReportSpdx = [(ReportSpdx, safeStripPrefix (show ReportSpdx) s)]
+  readsPrec _ s | s == show ReportMarkdown = [(ReportMarkdown, safeStripPrefix (show ReportMarkdown) s)]
+  readsPrec _ _ = []
```

Comment:
This is a little embarrassing, but the ONLY reason I suggested using `Read` at all was because I mistakenly thought that somehow it would derive from the `Show` instance (which, frankly, was silly).  Since that was incorrect, we should probably avoid using `Read` altogether (I'm so sorry for making you add this just to remove it).  It's just a bad typeclass overall, I really don't know why I thought this would be a good idea.

We should probably just make this a simple matching function `String -> Maybe ReportOutputFormat`.